spd/scripts/compare_models.py

```diff
@@ -0,0 +1,393 @@
+            return iter(
+                DatasetGeneratedDataLoader(
+                    dataset,
+                    batch_size=self.comparison_config.get("eval_batch_size", 1),
+                    shuffle=self.comparison_config.get("shuffle_data", False),
+                )
+            )
+
+        elif task_name == "lm":
+            from spd.data import DatasetConfig, create_data_loader
+
+            if "tokenizer_name" not in config or not config["tokenizer_name"]:
+                raise ValueError("tokenizer_name must be set for language models")
+
+            dataset_config = DatasetConfig(
+                name=task_config["dataset_name"],
+                hf_tokenizer_path=config["tokenizer_name"],
+                split=task_config["eval_data_split"],
+                n_ctx=task_config["max_seq_len"],
+                is_tokenized=task_config["is_tokenized"],
+                streaming=task_config["streaming"],
+                column_name=task_config["column_name"],
+                shuffle_each_epoch=task_config["shuffle_each_epoch"],
+                seed=None,
+            )
+            loader, _ = create_data_loader(
+                dataset_config=dataset_config,
+                batch_size=self.comparison_config.get("eval_batch_size", 1),
+                buffer_size=task_config["buffer_size"],
+                global_seed=config["seed"] + 1,
+                ddp_rank=0,
+                ddp_world_size=1,
+            )
+            return iter(loader)
+
+        elif task_name == "ih":
+            from spd.experiments.ih.model import InductionModelTargetRunInfo
+            from spd.utils.data_utils import DatasetGeneratedDataLoader, InductionDataset
+
+            if "pretrained_model_path" not in config or not config["pretrained_model_path"]:
+                raise ValueError("pretrained_model_path must be set for Induction Heads models")
+
+            target_run_info = InductionModelTargetRunInfo.from_path(config["pretrained_model_path"])
+            vocab_size = target_run_info.config.ih_model_config.vocab_size
+            seq_len = target_run_info.config.ih_model_config.seq_len
+            prefix_window = task_config.get("prefix_window") or seq_len - 3
+
+            dataset = InductionDataset(
+                vocab_size=vocab_size,
+                seq_len=seq_len,
+                prefix_window=prefix_window,
+                device=self.device,
+            )
+            return iter(
+                DatasetGeneratedDataLoader(
+                    dataset,
+                    batch_size=self.comparison_config.get("eval_batch_size", 1),
+                    shuffle=self.comparison_config.get("shuffle_data", False),
+                )
+            )
+
+        raise ValueError(
+            f"Unsupported task type: {task_name}. Supported types: tms, lm, resid_mlp, ih"
+        )
+
+    def compute_activation_densities(
+        self, model: ComponentModel, eval_iterator: Iterator[Any], n_steps: int = 5
```

Comment:
I'd remove the default n_steps: int = 5. I don't see the benefit when this isn't a public function that people will use from a lot of places.